en/material-system/effect-2.x-to-3.0.md

```diff
@@ -0,0 +1,254 @@
+
+    ```c
+    #include <shading-standard
+    #include <output-standard>
+    void surf (out StandardSurface s) {
+        // fill in your data here
+    }
+    vec4 frag () {
+        StandardSurface s; surf(s);
+        vec4 color = CCStandardShading(s);
+        return CCFragOutput(color);
+    }
+    ```
+
+    It is easy to implement custom surface input, or other shading algorithms in this framework.
+
+    > **Note**: the `CCFragOutput` function does not generally need to be implemented by itself, it only serves the purpose of interfacing with the rendering pipeline. And for outputs containing lighting calculations, the `output-standard` header should be included instead of `output` since the results are already in HDR range.
+
+- `CCToonShading`
+
+    Function name `CCToonShading`, which needs to contain the header file `shading-toon.chunk` for the lighting calculation for cartoon rendering.
+
+    ```c
+    #include <shading-toon
+    #include <output-standard>
+    void surf (out ToonSurface s) {
+        // fill in your data here
+    }
+    vec4 frag () {
+        ToonSurface s; surf(s);
+        vec4 color = CCToonShading(s);
+        return CCFragOutput(color);
+    }
+    ```
+
+### 3.3 Functions related to light and shadow calculation
+
+v2.x is very different from v3.0 in terms of light and shadow calculation, mainly including the following two parts:
+
+#### spherical light
+
+The **point light source** in v2.x is adjusted to **spherical light** in v3.0, and there are many ready-made functions, which need to be added to the header file `cc-forward-light.chunk` when using them.
+
+| Name | Type | Info |
+|:------ | :------ | :----- |
+| `cc_sphereLitPos[MAX_LIGHTS]` | vec4 |xyz: sphere light position |
+| `cc_sphereLitSizeRange[MAX_LIGHTS]` | vec4 | x: sphere light size<br>y: sphere light range |
+| `cc_sphereLitColor[MAX_LIGHTS]` | vec4 | xyz: sphere light color<br>w: sphere light intensity |
+
+For additional details, please refer to the [Common built-in shader Uniform](builtin-shader-uniforms.md) documentation.
+
+#### Spotlights
+
+The spotlight in v3.0 has a lot of ready-made features, which need to be added to the header file `cc-forward-light.chunk` when using it.
+
+| Name | Type | Info |
+| :----- | :----- | :----- |
+| `cc_spotLitPos[MAX_LIGHTS]` | vec4 | xyz: spotlight position |
+| `cc_spotLitSizeRangeAngle[MAX_LIGHTS]` | vec4 | x: spotlight size<br>y: spotlight range<br>z: spotlight angle |
+| vec4 | xyz: Spotlight direction |
+| `cc_spotLitColor[MAX_LIGHTS]` | vec4 | xyz: spotlight color<br>w: spotlight intensity |
+
+Please refer to the [Common shader built-in Uniform](builtin-shader-uniforms.md) documentation for additional details.
+
+### 3.4 Shadows
+
+There is a big difference between v2.x and v3.0 in shading calculations. v2.0 adds the header file `shadow.chunk`, while v3.0 adds the header file `cc-shadow.chunk`.
```

Comment:
There is a difference between v2.x and v3.0 in shadow calculations. In v2.0 you need to add the header file `shadow.chunk`, while in v3.0 the header file `cc-shadow.chunk` should be used instead.